statsmodels/stats/diagnostic.py

```diff
@@ -1611,8 +1619,14 @@ def recursive_olsresiduals(res, skip=None, lamda=0.0, alpha=0.95,
     rvarraw = np.nan * np.zeros(nobs)
 
     x0 = x[:skip]
+    if np.linalg.matrix_rank(x0) < x0.shape[1]:
+        err_msg = """\
+"The initial regressor matrix, x[:skip], issingular. You must use a value of
+skip large enough to ensure that the first OLS estimator is well-defined.
+"""
+        raise ValueError(err_msg)
```

Comment:
I don't think we need this, i.e. we use Ridge below with lamda > 0, or we can use pinv in line 1630.
For residuals we don't need to have identification of parameters.